File: python/27_Remove_Element.py
# -*- coding=utf-8 -*-
# -*- https://leetcode.com/problems/remove-element/ -*-
# -*- jasonlz -*-

class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        l = len(nums)
        if l==0:
            return
        i = 0
        while i < l:
            if nums[i]==val:
                nums[i]=nums[l-1]
                l = l-1
                continue
            i = i+1
        return l

File: C++/27_Remove_Element.cpp
/*https://leetcode.com/problems/remove-element/*/
/*jasonlz*/

class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
    	int len = nums.size();
    	if(len==0) return 0;
    	int i = 0;
    	while(i<len){
    		if(nums[i]==val){
    			nums[i]=nums[len-1];
    			len--;
    			continue;
    		}
    		i++;
    	}
    	return len;
    }
};